File: Backend API/CleaningHouse_API/Models/Wallet/WalletTransaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CleaningHouse_API.Models.Common;
using CleaningHouse_API.Models.Customers;

namespace CleaningHouse_API.Models.Wallet;

[Table("WalletTransactions")]
public class WalletTransaction
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int WalletId { get; set; }

    [Required]
    public int CustomerId { get; set; }

    public int? BookingId { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Amount { get; set; }

    [Required]
    [MaxLength(50)]
    public string Type { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    public string Direction { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    public string Status { get; set; } = WalletTransactionStatuses.Pending;

    [Required]
    [MaxLength(50)]
    public string PaymentMethod { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? ReferenceNumber { get; set; }

    [MaxLength(500)]
    public string? Notes { get; set; }

    public int? CreatedByAdminId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? CompletedAt { get; set; }

    [ForeignKey(nameof(WalletId))]
    public virtual Wallet? Wallet { get; set; }

    [ForeignKey(nameof(CustomerId))]
    public virtual AppUser? Customer { get; set; }

    [ForeignKey(nameof(BookingId))]
    public virtual Booking? Booking { get; set; }

    [ForeignKey(nameof(CreatedByAdminId))]
    public virtual AppUser? CreatedByAdmin { get; set; }
}

File: App_Clint/lib/features/wallet/domain/entities/wallet_transaction.dart
import 'package:equatable/equatable.dart';

/// Single wallet ledger row from GET /api/v1/wallet/transactions.
class WalletTransaction extends Equatable {
  final int id;
  final int walletId;
  final int customerId;
  final int? bookingId;
  final double amount;
  final String type;
  final String direction;
  final String status;
  final String? paymentMethod;
  final String? referenceNumber;
  final String? notes;
  final DateTime createdAt;
  final DateTime? completedAt;

  const WalletTransaction({
    required this.id,
    required this.walletId,
    required this.customerId,
    this.bookingId,
    required this.amount,
    required this.type,
    required this.direction,
    required this.status,
    this.paymentMethod,
    this.referenceNumber,
    this.notes,
    required this.createdAt,
    this.completedAt,
  });

  bool get isCredit => direction.toLowerCase() == 'credit';
  bool get isDebit => direction.toLowerCase() == 'debit';

  @override
  List<Object?> get props => [
        id,
        walletId,
        customerId,
        bookingId,
        amount,
        type,
        direction,
        status,
        paymentMethod,
        referenceNumber,
        notes,
        createdAt,
        completedAt,
      ];
}
